libbeat/common/stringset.go

```diff
@@ -49,3 +49,18 @@ func (set StringSet) Has(s string) (exists bool) {
 	}
 	return
 }
+
+// Equals compares this StringSet with another StringSet.
+func (set StringSet) Equals(anotherSet StringSet) bool {
+	if set.Count() != anotherSet.Count() {
+		return false
+	}
+
+	for k := range set {
```

Comment:
Hmm. This makes me wonder about the behavior of `Has()`. If it's just linear from L->R, then it seems like there is probably a point here where pre-sorting the sets will make this faster but I don't know if it's really worth it given that these sets are likely pretty small. 🤷‍♂ [Even the source code for `Search`](https://github.com/golang/text/blob/master/internal/stringset/set.go#L31) has "optimize" as a TODO so this is probably just fine as it is right now.

Reply:
The behavior of `Has` should be O(1) as it's looking up a key in a map. See https://github.com/elastic/beats/pull/12386/files#diff-15b17a3026eed8e21e4a3c4c6350d350R48.